src/modules/land_detector/VtolLandDetector.cpp

```diff
@@ -95,4 +95,13 @@ bool VtolLandDetector::_get_landed_state()
 	return landed;
 }
 
+bool VtolLandDetector::_get_freefall_state()
+{
+	bool low_gravity_detected = MulticopterLandDetector::_get_freefall_state(); // true if falling or in a parabolic flight
```

Comment:
I didn't see the log, but would it help if this was only called when fully in MC mode?

Basically thinking of `VtolLandDetector` is a combination of `FixedwingLandDetector` + `MulticopterLandDetector` that calls the appropriate implementation based on vtol state.

Reply:
@dagar That was my suggestion as well. I talked to @sfuhrer and his fix was meant to at least get rid of the false positives (free fall) during fixed wing mode.

VtolLandDetector would inherit from both `MulticopterLandDetector` and `FixedwingLandDetector`, correct?

Reply:
Given the relatively simple land detector interface and the need to selectively choose the implementation based on vtol state I'd probably do it with composition instead (VtolLandDetector contains a MulticopterLandDetector and FixedwingLandDetector). That way you'll also avoid the joys of c++ multiple inheritance.